### ISSUE TITLE
docs: add `ibis.example()` for an example dataset

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#5404]
### Is your feature request related to a problem?

Many of the [docstrings have examples](https://ibis-project.org/docs/4.1.0/api/expressions/tables/#ibis.expr.types.groupby.GroupedTable.mutate) that only use expressions that are unbound to data. It is not obvious to me what that groupby().mutate() funciton does from that. It would be nice if there was concrete data that we could see.

This would also be nice on the walkthroughs, eg in https://ibis-project.org/docs/4.1.0/ibis-for-pandas-users/ we download the iris dataset from github in the middle of the walkthrough, it would be nice if this worked offline.

### Describe the solution you'd like

[Vaex has `vaex.example()`](https://vaex.readthedocs.io/en/latest/index.html#Getting-started) that just gives you a dataset packaged with it, ready to go.

[sklearn does somethign similar](https://scikit-learn.org/stable/datasets/toy_dataset.html)

I think I might actually prefer sklearn's approach because it is better namespaced, and more extendable in the future. For example depending on the example you might want different datasets. Some datasets might be only ~100 rows and could be committed to git, so they are always available, while other larger ones could be fetched remotely.

I can write a PR if you give some guidelines.

### What version of ibis are you running?

target 5.0?

### What backend(s) are you using, if any?

all

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ogrisel** [comment]
Similarly, I think that many example snippets would benefit from display the concrete results of the execution and not just the schema of the expression (that is calling `.execute()`) in the example, assuming that operation is deterministic and does not risk crashing the docstests randomly.

**Comment by cpcloud** [comment]
> Similarly, I think that many example snippets would benefit from display the concrete results of the execution

Indeed. I did this for our [array APIs recently](https://ibis-project.org/api/expressions/collections/#ibis.expr.types.arrays.ArrayValue).

**Comment by cpcloud** [comment]
Re examples, what about something like this?

```python
ibis.read_example("name_of_example")   # return multiple tables if the example calls for it
```

**Comment by cpcloud** [comment]
`ibis.example("name_of_example")` would also be a nice API!

**Comment by jcrist** [comment]
For generated fake data sometimes it's nice to provide some configuration options to adjust the dimensionality of the data (nrows, ncols, proprtions, ...). What if instead we had a module (`ibis.examples`? `ibis.datasets`?) that provided some functions for retrieving/generating various datasets?

Scikit-learn breaks these into 3 categories:

- `sklearn.datasets.load_*` loads data shipped with scikit-learn
- `sklearn.datasets.fetch_*` fetches external data
- `sklearn.datasets.make_*` generates fake data

So something like

```python
t = ibis.datasets.load_iris()
```

Making these individual functions also allows for possibly configuring additional things like the backend used.

**Comment by cpcloud** [comment]
`load_*` seems good to me, I wonder if we can elide the `fetch_*`-style since external vs internal should be something that we can abstract over.

**Comment by jcrist** [comment]
Yeah. I wasn't suggesting we necessarily adopt this naming convention, although it might be nice just to indicate which examples are "cheap" vs "possibly expensive" vs auto-generated :shrug:.

**Comment by cpcloud** [comment]
True, that is a nice breakdown of expectations. Ok, maybe we just copy that convention then!

**Comment by Holer90** [comment]
I second the suggestion to follow the syntax of Scikit-learn for loading baked-in datasets.

**Comment by NickCrews** [comment]
I like how succinct `ibis.example("name")` is, and I would vote for that, except it has the downside that it isn't tab-autocomplete friendly. I would love to go `ibis.datasets.<TAB>` and have it suggest things for me, ideally with a docstring (ie I don't need to go the web docs) that says
- short description eg "50 samples from 3 species of flowers. Includes 4 columns of measurements and 1 of the species label"
- nrows
- column description
- approx size in kb, Mb, Gb
- whether baked in or needs to fetched

I like each dataset being namespaced in a subpackage. No preference between examples/datasets. No preference for including/leaving out the load_/fetch_/make_ prefixes.

**Comment by NickCrews** [comment]
Also, I'd imagine we'd store the data in `.sqlite` files and these functions would return the result of `ibis.sqlite.connect(DATA_PATH)`, ready to run some computation?
